execution-core/src/transfer.rs

```diff
@@ -285,6 +285,17 @@ impl Transaction {
         })
     }
 
+    /// Returns gas charge for the deployment of a contract bytecode.
+    /// This will be zero if the transaction is not a deploy transaction.
+    #[must_use]
+    pub fn deploy_charge(&self, gas_per_deploy_byte: u64) -> u64 {
```

Comment:
I am surprised seeing  a deploy charge calculating function inside execution core.
Are we sure this is the right place?

Reply:
It is a method on the transaction, where else would you think it makes sense?

Reply:
The pro of having it inside execution core is to let clients (wallets) to be able to calculate the required fee (simply asking the gas_per_deploy_byte parameter to the node)

Reply:
> The pro of having it inside execution core is to let clients (wallets) to be able to calculate the required fee (simply asking the gas_per_deploy_byte parameter to the node)

I honestly like it a lot. We currently have a custom implementation in Rusk wallet, we might have some implementations in tests, we will need an implementation for the Web wallet. Better to centralize the behavior in one spot rather than having it scattered around, right?

Reply:
Problem is that such function should really only be in one place, and in its current form here it is incorrect (gives incorrect result) as it does not take into account `min_deploy_points` configuration item, which increases the charge for small contracts. So the duplication leads to incorrectness. To approve this PR we either need to correct it (add a min_deploy_points argument) or to eliminate it.

Reply:
I have rebased the PR on top of the `min_deploy_points` changes and it's ready again for review